lib/projectPreprocessor.js

```diff
@@ -422,17 +421,23 @@ class ProjectPreprocessor {
 		}
 
 		const validationResults = await Promise.all(
-			configs.map((config, documentIndex) => validate({
-				config,
-				project: {
-					id: project.id
-				},
-				yaml: {
-					path: configPath,
-					source: configFile,
-					documentIndex
+			configs.map(async (config, documentIndex) => {
+				try {
```

Comment:
We should document that the try/catch block here is required to get the first error of the first document. And not the first error to occur in any document.

Reply:
Done